File: EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/EducationalSupervisionVisit.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class EducationalSupervisionVisit : BaseAuditableEntity
{
    public long DirectorateId { get; set; }
    public long SchoolId { get; set; }
    public string SupervisorName { get; set; } = string.Empty;
    public DateTime VisitDate { get; set; }
    public string VisitPurpose { get; set; } = string.Empty;
    public decimal? EvaluationScore { get; set; }
    public string? Recommendations { get; set; }
    public int Status { get; set; } // 1=Scheduled, 2=Completed, 3=FollowUpRequired
    public long? SupervisorEmployeeId { get; set; }
    public long? TargetDepartmentId { get; set; }
    public DateTime? FollowUpRequiredDate { get; set; }
    public string? ActionItemsDetail { get; set; }

    // Navigation Properties
    public virtual Directorate? Directorate { get; set; }
    public virtual School? School { get; set; }
}

File: EduMS-Frontend/edums-frontend-DDD-System/src/app/modules/school-admin/interfaces/educational-supervision-visit.ts
import type { VisitStatus } from './common';

export interface EducationalSupervisionVisit {
  id: number;
  directorateId: number;
  schoolId: number;
  supervisorName: string;
  visitDate: string;
  visitPurpose: string;
  evaluationScore: number | null;
  recommendations: string | null;
  status: VisitStatus;
  supervisorEmployeeId: number | null;
  targetDepartmentId: number | null;
  followUpRequiredDate: string | null;
  actionItemsDetail: string | null;
}

export type CreateEducationalSupervisionVisitDto = Omit<EducationalSupervisionVisit, 'id' | 'status'>;

export type UpdateEducationalSupervisionVisitDto = Omit<EducationalSupervisionVisit, 'directorateId' | 'schoolId' | 'status'>;
